### PR TITLE
docs: refactor installation docs around Docker (DOCS-3009)

### PR DESCRIPTION
**Pull request by JimGalasyn**
Rewrites the installation docs to focus on Docker containers.

### CODE REVIEW

File: docs-md/operate-and-deploy/installation/install-ksqldb-with-docker.md [hunk 1]
```diff
@@ -142,13 +121,13 @@ docker run -d \
 In interactive mode, a ksqlDB CLI instance running outside of Docker can
 connect to the ksqlDB server running in Docker.
 
-Connect ksqlDB Server to a Secure Kafka Cluster, Like Confluent Cloud
-===================================================================
+### Connect ksqlDB Server to a Secure Kafka Cluster, Like Confluent Cloud
```
Comment (JoyceFee1) [inline docs-md/operate-and-deploy/installation/install-ksqldb-with-docker.md]: I wouldn't capitalize "Like" here, but up to you.

File: docs-md/operate-and-deploy/installation/installing.md [hunk 2]
```diff
@@ -3,144 +3,222 @@ layout: page
...
-| --------------------- | ------------------ |
-| Apache Kafka version  | 0.11.0 and later   |
-| {{ site.cp }} version | > 3.3.0 and later  |
+Stacks that have {{ site.sr }} can use Avro-encoded events in ksqlDB
+applications. Without {{ site.sr }}, your ksqlDB applications can use only JSON
+or delimited formats.  
 
-Installation Instructions
--------------------------
+!!! note
```
Comment (JoyceFee1) [inline docs-md/operate-and-deploy/installation/installing.md]: I cannot find a way to view the rendered output (no link in console output), so cannot view how this `!!! note` renders. Hoping it looks like a normal note and not three exclamation marks with the text.

File: docs-md/operate-and-deploy/installation/installing.md [hunk 3]
```diff
@@ -3,144 +3,222 @@ layout: page
...
+
+Create a stack file to define your ksqlDB application 
+-----------------------------------------------------
+
+When you've decided on the services that you want in the stack, you define a
+[Compose file, or "stack" file](https://docs.docker.com/compose/compose-file/),
+which is a YAML file, to configure your ksqlDB application's services. The 
+stack file is frequently named `docker-compose.yml`. 
+
+To start your ksqlDB application, you use the
```
Comment (JoyceFee1) [inline docs-md/operate-and-deploy/installation/installing.md]: "To start your ksqlDB application, use the..."

File: docs-md/operate-and-deploy/installation/installing.md [hunk 4]
```diff
@@ -3,144 +3,222 @@ layout: page
...
+[docker-compose CLI](https://docs.docker.com/compose/) to
```
Comment (JoyceFee1) [inline docs-md/operate-and-deploy/installation/installing.md]: ...run the stack for the application. Run `docker-compose up` to start the
application and...

File: docs-md/operate-and-deploy/installation/installing.md [hunk 5]
```diff
@@ -3,144 +3,222 @@ layout: page
...
+run the stack for your application. Run `docker-compose up` to start your
+application and `docker-compose down` to stop it.
+
+!!! note
+    If your stack file is compatible with version 3 or higher,
```
Comment (JoyceFee1) [inline docs-md/operate-and-deploy/installation/installing.md]: As you can tell from previous comments, use of "your/your"  can be replaced by "the"

File: docs-md/operate-and-deploy/installation/installing.md [hunk 6]
```diff
@@ -3,144 +3,222 @@ layout: page
...
+convention is named `docker-compose.yml`.
+
+### 3. Bring up the stack and run ksqlDB
+   
+To bring up the stack and run ksqlDB, use the
+[docker-compose](https://docs.docker.com/compose/reference/overview/) tool,
+which reads your `docker-compose.yml` file and runs containers for your
+{{ site.ak }} and ksqlDB services. 
+
+ksqlDB Tutorial stack
```
Comment (JoyceFee1) [inline docs-md/operate-and-deploy/installation/installing.md]: curious why cap used in "Tutorial" here...notice you are using sentence style cap.

File: docs-md/operate-and-deploy/installation/installing.md [hunk 7]
```diff
@@ -163,15 +241,229 @@ Having trouble? Type 'help' (case-insensitive) for a rundown of how things work!
...
+Stopping tutorials_kafka_1           ... done
+Stopping tutorials_zookeeper_1       ... done
+Removing tutorials_ksql-server_1     ... done
+Removing tutorials_schema-registry_1 ... done
+Removing tutorials_kafka_1           ... done
+Removing tutorials_zookeeper_1       ... done
+Removing network tutorials_default
+```
+
+Specify ksqlDB Server configuration parameters
```
Comment (JoyceFee1) [inline docs-md/operate-and-deploy/installation/installing.md]: ah--so are ksqlDB Server and ksqlDB Tutorial actual product names?

File: docs-md/operate-and-deploy/installation/installing.md [hunk 8]
```diff
@@ -163,15 +241,229 @@ Having trouble? Type 'help' (case-insensitive) for a rundown of how things work!
...
+container for ksqlDB Server. For more information, see
+[Configuring ksqlDB Server](server-config/index.md).
 
-You can use ksqlDB with a {{ site.ak }} cluster in {{ site.ccloud }}. For more
-information, see
+To start ksqlDB containers in configurations like "ksqlDB Headless Server"
+and "ksqlDB Interactive Server (Development)", see
+[Configure ksqlDB with Docker](install-ksqldb-with-docker.md).
+
+Supported Versions and Interoperability
```
Comment (JoyceFee1) [inline docs-md/operate-and-deploy/installation/installing.md]: Supported versions and interoperability?

File: docs-md/operate-and-deploy/installation/installing.md [hunk 9]
```diff
@@ -163,15 +241,229 @@ Having trouble? Type 'help' (case-insensitive) for a rundown of how things work!
...
+
+You can use ksqlDB with compatible {{ site.aktm }} and {{ site.cp }}
+versions.
+
+|    ksqlDB version     | {{ site.release }} |
+| --------------------- | ------------------ |
+| Apache Kafka version  | 0.11.0 and later   |
+| {{ site.cp }} version | > 3.3.0 and later  |
+
+Scale Your ksqlDB Server Deployment
```
Comment (JoyceFee1) [inline docs-md/operate-and-deploy/installation/installing.md]: Scale ksqlDB Server deployment?

File: docs-md/operate-and-deploy/installation/installing.md [hunk 10]
```diff
@@ -163,15 +241,229 @@ Having trouble? Type 'help' (case-insensitive) for a rundown of how things work!
...
+and the CLI connects to these remote servers.
+
+![image](../../img/client-server.png)
+
+You can add or remove servers from the same resource pool during live
+operations, to scale query processing. You can use different resource pools
+to support workload isolation. For example, you could deploy separate pools
+for production and for testing.
+
+Next Steps
```
Comment (JoyceFee1) [inline docs-md/operate-and-deploy/installation/installing.md]: Next steps?

File: docs-md/operate-and-deploy/installation/installing.md [hunk 11]
```diff
@@ -163,15 +241,229 @@ Having trouble? Type 'help' (case-insensitive) for a rundown of how things work!
...
+file shows how to configure a stack with these features:
+
+- Start a {{ site.ak }} cluster with two brokers.
+- Start a {{ site.kconnect }} instance.
+- Start {{ site.sr }}. 
+- Start containers running Elasticsearch and Kibana.
+- Start ksqlDB Server and ksqlDB CLI containers.
+
+!!! note
+    You need to install
```
Comment (JoyceFee1) [inline docs-md/operate-and-deploy/installation/installing.md]: You must?

**Comment by JimGalasyn** [comment]
Thanks for the review, @JoyceFee1!

File: docs-md/operate-and-deploy/installation/installing.md (hunk 2 above)
Comment (JimGalasyn) [reply docs-md/operate-and-deploy/installation/installing.md]: Yes, it'll only render with `mkdocs`, but I doubt it's worth the effort for you to install it. :)

**Pull request merged**